precompiles/utils/src/data.rs

```diff
@@ -214,7 +226,18 @@ impl EvmData for Address {
 	}
 }
 
-impl EvmData for U256 {
+/// A local marker trait for types that are TryFrom<U256> so we can have a blanket impl
+/// This is necessary to avoide duplicate trait implementation errors.
+/// See https://github.com/PureStake/moonbeam/pull/518#discussion_r682493750 for details
+pub trait EvmDataTryFromU256Marker: TryFrom<U256> + Into<U256> {}
+impl EvmDataTryFromU256Marker for u8 {}
+impl EvmDataTryFromU256Marker for u16 {}
+impl EvmDataTryFromU256Marker for u32 {}
+impl EvmDataTryFromU256Marker for u64 {}
+impl EvmDataTryFromU256Marker for u128 {}
+impl EvmDataTryFromU256Marker for U256 {}
+
+impl<T: EvmDataTryFromU256Marker> EvmData for T {
```

Comment:
It works well but I think it's not the most efficient, because the integer is first transformed into U256 before being read or written. It would be more efficient to read/write the integer directly, as it is done with bool. But this would require to make a specific impl for each integer (can be factorizable via a macro).